bloges56/DateSim
# Set Name Scene Addition

## Description
- There is now a scene to set your name before the game takes you to the arcade. This scene should load after pressing the start button

## How to Test
1. Go to SceneManager Scene and hit play
3. Hit start game
4. Name scene should load. 
5. Enter your name. 
6. Press continue or return/enter on your keyboard to go to the next scene 
7. Arcade scene should load. 
8. Press B on your keyboard. The name you entered should be printed in the console. 
9. Continue playing normally. All other things should be unchanged. 


## Checklist
- [x] Does the name scene load after pressing start? 
- [x] Can you type your name when you click on the input prompt?   
- [x] Does pressing continue take you to the Arcade? 
- [x] Does pressing return/enter key take you to the arcade?
- [x] Can you see the name you entered when you press B in the arcade?
- [x] Does everything else in the arcade work as intended?  

## Key Changes
1. [Scene Added]SetName scene has been added.
2. [Global Variable Added] The name is now stored in the game manager. You should be able to access it from anywhere. See Usage below for instructions on how to access it. 



## Usage
1. Add the Managers prefab to your scene— if its not already there. 
2. Use the managers in your own script— see HomeScreenManager.cs for a reference: 

	Initialize your variables to store the GameManager and any other managers you create. 
	``` 
    GameManager gameManager;
	```
	In the Start() function, assign references to each manager through the Manager prefab as follows: 
	```        
    gameManager = Managers.gameManager;
    ```
    Use the managers in your code!

3. After adding the manager in your own script, you can access the name as follows: 
	```
    Debug.Log("Name is: "+gameManager.GetName());
    ```
    You can see how it is used inside of the NameManager.cs script on line 51. 

4. If you need to update the name, you can do so with the following code: 
	```
	gameManager.SetName("Name");
	```
	You can see how its used inside of the NameManager.cs script on line 55.

5. In order to work on the dialogue, you can set the askForName variable to false in the NameManager.cs script on line 21. This will hide the input form so you can start the dialogue. 
Once your dialogue asks for the name, set askForName to true. If you want to add dialogue after setting the name, remove the call to `DoneName()` on line 58. Once you're finished with the dialogue, add a call to `DoneName()`. If you need help, feel free to reach out to me!
